pandas/core/common.py

```diff
@@ -1321,6 +1321,8 @@ def _possibly_downcast_to_dtype(result, dtype):
             try:
                 result = result.astype(dtype)
             except:
+                # convert to datetime and change timezone
```

Comment:
this is too general. it can be a datetimelike (e.g. either M or m), and may or may not have a timezone. I think more tests are in order.